onsp_co2/models/carbon_factor.py

```diff
@@ -54,6 +54,8 @@ class CarbonFactor(models.Model):
         related="recent_value_id.carbon_monetary_currency_id"
     )
     unit_label = fields.Char(related="recent_value_id.unit_label")
+    
+    carbon_factor_type_ids = fields.Many2many("carbon.factor.type", string="Carbon Factor Type Required")
```

Comment:
I know that I put a lot of `carbon_` prefixes in fields, but I now try to remove them on models that already start with a `carbon.` prefix.
Something like `type_ids` should be enough and that will make the code shorter where the field is used.
Also, we could rename it to `required_type_ids` as it is exactly that :)

Maybe the string should be better also?